### ISSUE TITLE
Warning FS3389 for implicit number widening does not work

### ISSUE THREAD

**Issue by hyazinthh** [dotnet/fsharp#12496]
Turning on warning FS3389 for implicit number widening does not seem to have any effect. Got the warning codes from https://docs.microsoft.com/en-us/dotnet/fsharp/whats-new/fsharp-6#optional-warnings-for-implicit-conversions

**Repro steps**
```fsharp
#warnon "3388"
#warnon "3389"
#warnon "3391"
#warnon "3395"

let foo (value : int64) =
    ()

let data : int64[] = [| 1; 2; 3 |]
foo 0
```
Am I missing something here? Other warnings (e.g. FS3391) seem to work as intended.

[Test.zip](https://github.com/dotnet/fsharp/files/7667251/Test.zip)

**Expected behavior**

Should trigger a warning about the implicit number widening.

**Actual behavior**

No warning is produced.

**Related information**

* VS 2022
* .NET / F# 6

**Comment by dsyme** [comment]
`#warnon "NNNN"` is not recognised in scripts:

```
  #warnon "3389"
  ^^^^^^^^^^^^^^

stdin(2,1): warning FS3353: Invalid directive '#warnon 3389'
```

Using 
```
fsi --warnon:3389
```

I get
```txt
  let data : int64[] = [| 1; 2; 3 |]
  ------------------------^

stdin(56,25): warning FS3389: This expression uses a built-in implicit conversion to convert type 'int' to type 'int64'. See https://aka.ms/fsharp-implicit-convs.
```

**Comment by dsyme** [comment]
Please try again and reopen with more detailed exact version/repro steps if this persists, thanks

**Comment by hyazinthh** [comment]
Thanks for the quick reply.

You are right! My bad, at first I tried adding `<WarnOn>` to a project file directly, which did not produce a warning. Just tried it again and it worked as expected. I guess there must have been a typo.

I'm not sure if this is the right place to mention this, but the article I linked above says:
> You may optionally enable the warning /warnon:3395 to show a warning at every point implicit numeric widening is used, as described in Optional warnings for implicit conversions.

This should say /warnon:3389 instead, shouldn't it?
